osgp/platform/osgp-adapter-domain-smartmetering/src/main/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/mapping/customconverters/EventsConverter.java

```diff
@@ -34,7 +39,12 @@ public EventDto convertFrom(final Event source, final Type<EventDto> destination
             return null;
         }
 
-        return new EventDto(source.getTimestamp(), source.getEventCode(), source.getEventCounter(),
-                source.getEventLogCategory().name());
+        if (source.getEventLogCategory() == EventLogCategory.POWER_FAILURE_EVENT_LOG) {
+            return new EventDto(source.getTimestamp(), source.getEventCode(), source.getEventLogCategory().name(),
```

Comment:
why not make one constructor and let the startTime and Duration be null? In that case you can skip the IF statement

Reply:
Dat is niet zo netjes

Reply:
done it